src/test/java/org/apache/sling/scripting/sightly/impl/engine/extension/FormatFilterExtensionTest.java

```diff
@@ -99,47 +100,26 @@ public void testDateFormatWithZoneName() {
     }
 
     @Test
-    public void testDateFormatWithEscapedCharactersJdk8() {
-        assumeJdk8();
-        assertDate("01 December '18 12:00 AM; day in year: 335; week in year: 49",
-            "dd MMMM ''yy hh:mm a; 'day in year': D; 'week in year': w",
-            "UTC",
-            null);
-    }
-
-    @Test
-    public void testDateFormatWithEscapedCharactersJdk11OrNewer() {
-        assumeJdk11OrNewer();
+    @Ignore("ambiguous results depending on the jdk version and implementation used")
```

Comment:
Isn't this rather depending on the Locale Provider? If you set this explicitly to CLDR (https://openjdk.java.net/jeps/252) it should work with all distributions IMHO, compare with https://issues.apache.org/jira/browse/SLING-9984 and https://docs.huihoo.com/java/javase/9/docs/api/java/util/spi/LocaleServiceProvider.html.

Reply:
We rely on stable output also in the HTL Integration tests

Reply:
Yes setting it explicitly to *COMPAT* would work, but only when you run the tests from maven. If you run them in the ide you have to configure the test runner as well accordingly. It should be good enough to leave a comment at the tests for that case.

Reply:
If the tests run reliably from Maven I'm fine, and as you mention a comment for IDEs is good enough. But I don't think the tests should be @ignored by default, my view is that Maven must run all tests reliably, and IDEs come second.

Reply:
I consider `CLDR` the more compatible solution across different JDK distributions, therefore please rely on that whenever possible and set that as preferred option via https://docs.oracle.com/javase/8/docs/api/java/util/Properties.html#setProperty-java.lang.String-java.lang.String-. That way it should work both from IDE and from Maven. As long as tests are not executed in parallel it should be safe to set to `CLDR` prior to your tests and reset it after them.

Reply:
COMPAT is used here as well https://github.com/apache/sling-org-apache-sling-scripting-sightly-testing/blob/09b67343e22d06db7cbe01e8c3e8342df1d01d63/pom.xml#L194 

It should actually be `COMPAT,SPI` according to https://www.oracle.com/java/technologies/javase/java9locales.html 

> Specifying the system property with "COMPAT,SPI" will have the same behavior with the prior JDK releases.

`CLDR` is not reliable as it depends on the version used by the particular implementation java.net jdk11.0.2 gives different results then the oracle jdk11 gives. 

The configuration cannot be changed on runtime 
> The search order of locale sensitive services can be configured by using the "java.locale.providers" system property. This system property declares the user's preferred order for looking up the locale sensitive services separated by a comma. It is only read at the Java runtime startup, so the later call to System.setProperty() won't affect the order.

Reply:
Maybe we should add the change to the surefire plugin configuration to the parent pom https://github.com/apache/sling-org-apache-sling-scripting-sightly/pull/11/commits/586ff8048c578d0ac6aa2e632f4745bc2294e746

Reply:
> COMPAT is used here as well https://github.com/apache/sling-org-apache-sling-scripting-sightly-testing/blob/09b67343e22d06db7cbe01e8c3e8342df1d01d63/pom.xml#L194

This is for executing the HTL TCK (https://github.com/adobe/htl-tck) which cannot be changed easily. You are right that this change only fixes compatibility with Oracle JDKs If you require the TCP with being compatible with other JDKs then it is much more effort.

For local Sling test we are free to define Tests in a more compatible way which is IMHO CLRD.

Reply:
When using `CLDR` the tests work on oracle jdk8 and fail on oracle jdk11 and any other jdk11. If using `COMPAT` works for both, so imho we should keep it like that.

Reply:
That is interesting. Can you share the output (with concrete versions)? Is CLDR really so bad with backwards compatibility or was it just implemented incorrectly in Oracle JDK8?

Reply:
It was wrong in the JDK8 I guess. Btw not sure if it is intent but Jenkins builds with jdk8 and runs sonar with jdk11. 

short / GMT+02:00 / fr 
==> jdk8u301 `01/12/18` vs jdk11.0.12 `01/12/1918`
EEEE, d MMM y / UTC / de 
==> jdk8u301 `Sonntag, 1 Dez 1918` vs jdk11.0.12 `Sonntag, 1 Dez. 1918`
dd MMMM ''yy hh:mm a; 'day in year': D; 'week in year': w / UTC 
==> jdk8u301 `01 December '18 12:00 AM; day in year: 335; week in year: 49`
==> jdk11.0.12 `01 December '18 12:00 AM; day in year: 335; week in year: 48`

Reply:
I did also some more tests, JDK8 (Zulu, AdoptOpenJdk and Oracle) ships with CLDR v27, while JDK11 ships with CLDR v33 (again all three distributions). The differences are quite fundamental (surprisingly) for instance https://bugs.openjdk.java.net/browse/JDK-8211262 or the year using 4 digits instead of 2 in short format style for most locales.